### PR TITLE
Bump up envoyproxy control-plane to 0.9

### PR DESCRIPTION
**Pull request by prasadborole1**
**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [X] Documentation updated?

**Affected functionality**

**Description of change**

Upgrading the envoyproxy/go-control-plane dependency version from 0.8 to 0.9.
This version uses golang/protobuf (https://github.com/envoyproxy/go-control-plane/releases/tag/v0.9.0). 
The commit has changes made to sds handler as per updated interfaces.

### CODE REVIEW

File: pkg/agent/endpoints/sds/handler.go [hunk 1]
```diff
@@ -54,6 +55,10 @@ type Handler struct {
 	}
 }
 
+func (h *Handler) DeltaSecrets(discovery_v2.SecretDiscoveryService_DeltaSecretsServer) error {
```
Comment (evan2645) [inline pkg/agent/endpoints/sds/handler.go]: Please define this below NewHandler with the rest of the functions that have a *Handler receiver

File: pkg/agent/endpoints/sds/handler.go [hunk 2]
```diff
@@ -54,6 +55,10 @@ type Handler struct {
...
+	return errors.New("not implemented")
```
Comment (evan2645) [inline pkg/agent/endpoints/sds/handler.go]: Should this be gRPC error code UNIMPLEMENTED ?

File: pkg/agent/endpoints/sds/handler.go (hunk 1 above)
Comment (prasadborole1) [reply pkg/agent/endpoints/sds/handler.go]: sure

File: pkg/agent/endpoints/sds/handler.go (hunk 2 above)
Comment (prasadborole1) [reply pkg/agent/endpoints/sds/handler.go]: Updated

**Pull request merged**